packages/internal/src/storage-test-utils.js

```diff
@@ -6,6 +6,18 @@ import { bindAllMethods } from './method-tools.js';
 
 const { Fail, quote: q } = assert;
 
+/**
+ * A map corresponding with a total function such that `get(key)`
+ * is assumed to always succeed.
+ *
+ * @template K, V
+ * @typedef {{[k in Exclude<keyof Map<K, V>, 'get'>]: Map<K, V>[k]} & {get: (key: K) => V}} TotalMap
+ */
+/**
+ * @template T
+ * @typedef {T extends Map<infer K, infer V> ? TotalMap<K, V> : never} TotalMapFrom
+ */
```

Comment:
I wrote these because I got sick of TypeScript complaining that the result of `map.get(key)` can be undefined in circumstances where we know it won't. They should probably migrate to internal or endo at some point.

Reply:
:clap: